src/erk/cli/commands/implement.py

```diff
@@ -434,51 +437,33 @@ def _prepare_plan_source_from_issue(
     # Output issue title
     ctx.feedback.info(f"Issue: {plan.title}")
 
-    # Validate issue has erk-plan label
-    if "erk-plan" not in plan.labels:
-        user_output(
-            click.style("Error: ", fg="red")
-            + f"Issue #{issue_number} does not have the 'erk-plan' label.\n"
-            + "Only issues tagged with 'erk-plan' can be used for implementation.\n\n"
-            + f"To add the label, visit: {plan.url}"
-        )
-        raise SystemExit(1)
-
-    # Validate issue is open
-    if plan.state != PlanState.OPEN:
-        user_output(
-            click.style("Warning: ", fg="yellow")
-            + f"Issue #{issue_number} is {plan.state.value}. "
-            + "Proceeding anyway..."
-        )
+    # Validate using shared helper
+    try:
+        warnings = validate_plan_for_worktree(plan)
+        for warning in warnings:
+            user_output(click.style("Warning: ", fg="yellow") + warning)
+    except IssueValidationError as e:
+        user_output(click.style("Error: ", fg="red") + str(e))
+        raise SystemExit(1) from None
 
-    # Create branch name: P prefix + issue number + sanitized title + timestamp
-    # Apply P prefix AFTER sanitization since sanitize_worktree_name lowercases input
-    # Truncate total to 31 chars before adding timestamp suffix
-    prefix = f"P{issue_number}-"
-    sanitized_title = sanitize_worktree_name(plan.title)
-    base_branch_name = (prefix + sanitized_title)[:31].rstrip("-")
-    timestamp_suffix = format_branch_timestamp_suffix(ctx.time.now())
-    branch_name = base_branch_name + timestamp_suffix
+    # Prepare branch/worktree names using shared helper
+    setup = prepare_plan_for_worktree(plan, ctx.time.now())
```

Comment:
I think we should structure this so that the logic for validate_plan_for_worktree is folded INTO prepare_plan_for_worktree and we should handle the different cases that clients have to deal with using union types